### PR TITLE
Faces.js and Maven compressor

### PR DESCRIPTION
**Pull request by pizzi80**
As promised a long time ago, this is my contribution to Mojarra:
This is a first release which can be used as a base for further code cleanups and optimizations.

### Maven

1.  Removed yuicompressor (dead project with no support for ES6+ js)
2.  Added Google Closure maven plugin for minification 
     https://github.com/blutorange/closure-compiler-maven-plugin

### Faces JS

1. tested and minified with ECMASCRIPT5_STRICT profile of Google Closure
2. use strict
3. const and let instead of var
4. Removed IE-5-6-7-8-9-10 support 
5. Removed IE 5-6-7 hacks and memory hacks
6. Removed FrameTransport
7. FormData for File upload
8. Many minor optimizations 
9. Removed / Commented out unused and very old functions
10. ~ 10 kB Minified and Gzipped

#5045 
#5040

sorry if I made some mistake with PR or similar... I'm not used to contribute to open source github projects

### CODE REVIEW

**Comment by pizzi80** [comment]
https://github.com/jakartaee/faces/issues/1598

**Comment by pizzi80** [comment]
> This looks great but it looks like you need to sign the legal agreement and then watch this video how to fix this PR with your signature! https://www.youtube.com/watch?v=YdJyJpT7rrs

I've created an Eclispe account wired with this github account ... I need to follow the instructions on the video... ;)

File: impl/src/main/resources/META-INF/resources/jakarta.faces/faces-uncompressed.js [hunk 1]
```diff
@@ -1127,139 +672,127 @@ if (!((faces && faces.specversion && faces.specversion >= 23000 ) &&
...
                     //name when using getAttribute.
-                    if( attributeName == 'class'){
-                        if( isIE() && (source.getAttribute(propertyName) === source[propertyName]) ){
-                            attributeName = propertyName;
-                        }
+                    // if( attributeName === 'class'){
+                    //     if( isIE() && (source.getAttribute(propertyName) === source[propertyName]) ){
+                    //         attributeName = propertyName;
+                    //     }
+                    // }
```
Comment (melloware) [inline impl/src/main/resources/META-INF/resources/jakarta.faces/faces-uncompressed.js:725]: Looks like extra comment here can be removed?

File: impl/src/main/resources/META-INF/resources/jakarta.faces/faces-uncompressed.js [hunk 2]
```diff
@@ -1127,139 +672,127 @@ if (!((faces && faces.specversion && faces.specversion >= 23000 ) &&
...
-                if (oldBooleanValue != newBooleanValue) {
+                // } else {
+                //     //setting property to '' seems to be the only cross-browser method for removing an attribute
+                //     //avoid setting 'value' property to '' for checkbox and radio input elements because then the
+                //     //'value' is used instead of the 'checked' property when the form is serialized by the browser
+                //     if (attributeName === "value" && (target.type !== 'checkbox' && target.type !== 'radio')) {
+                //          target[propertyName] = '';
+                //     }
+                //     target.removeAttribute(attributeName);
+                // }
```
Comment (melloware) [inline impl/src/main/resources/META-INF/resources/jakarta.faces/faces-uncompressed.js:740]: Same here?

File: impl/src/main/resources/META-INF/resources/jakarta.faces/faces-uncompressed.js [hunk 3]
```diff
@@ -1293,25 +826,28 @@ if (!((faces && faces.specversion && faces.specversion >= 23000 ) &&
...
+
+            //if (typeof DOMParser !== "undefined") {  // FF, S, Chrome
+            //    doc = (new DOMParser()).parseFromString(docStr, "text/xml");
+            //}
+            // else if (typeof ActiveXObject !== "undefined") { // IE
+            //     doc = new ActiveXObject("MSXML2.DOMDocument");
+            //     doc.loadXML(docStr);
+            // } else {
+            //     throw new Error("You don't seem to be running a supported browser");
+            // }
```
Comment (melloware) [inline impl/src/main/resources/META-INF/resources/jakarta.faces/faces-uncompressed.js:844]: Remove commented code.

File: impl/src/main/resources/META-INF/resources/jakarta.faces/faces-uncompressed.js [hunk 4]
```diff
@@ -1325,19 +861,27 @@ if (!((faces && faces.specversion && faces.specversion >= 23000 ) &&
...
-                for (var i = 0, length = formElements.length; i < length; i++) {
-                    var formElement = formElements[i];
+                // const formElements = form.elements;
+                // let i = 0, length = formElements.length;
+                // for (; i < length; i++) {
+                //     const formElement = formElements[i];
+                //     if (formElement.name && (formElement.name.indexOf('jakarta.faces.encodedURL') >= 0)) {
+                //         return formElement;
+                //     }
+                // }
```
Comment (melloware) [inline impl/src/main/resources/META-INF/resources/jakarta.faces/faces-uncompressed.js:877]: Remove commented code.

File: impl/src/main/resources/META-INF/resources/jakarta.faces/faces-uncompressed.js [hunk 5]
```diff
@@ -1346,51 +890,55 @@ if (!((faces && faces.specversion && faces.specversion >= 23000 ) &&
...
-                    var formElement = formElements[i];
-
+                // const formElements = form.elements;
+                // let i = 0, length = formElements.length;
+                // for (; i < length; i++) {
+                //     const formElement = formElements[i];
+                //     if (formElement.name && (formElement.name.indexOf(hiddenStateFieldName) >= 0)) {
+                //         return formElement;
+                //     }
+                // }
```
Comment (melloware) [inline impl/src/main/resources/META-INF/resources/jakarta.faces/faces-uncompressed.js:939]: Remove commented code

File: impl/src/main/resources/META-INF/resources/jakarta.faces/faces-uncompressed.js [hunk 6]
```diff
@@ -1487,32 +1036,35 @@ if (!((faces && faces.specversion && faces.specversion >= 23000 ) &&
...
+                    const tableElements = ['td', 'th', 'tr', 'tbody', 'thead', 'tfoot'];
+                    const isInTable = tableElements.includes(tag);
+                    // let isInTable = false;
+                    // let tei = 0, tel = tableElements.length;
+                    // for (; tei < tel; tei++) {
+                    //     if (tableElements[tei] === tag) {
+                    //         isInTable = true;
+                    //         break;
+                    //     }
+                    // }
```
Comment (melloware) [inline impl/src/main/resources/META-INF/resources/jakarta.faces/faces-uncompressed.js:1064]: Remove commented code.

**Review by melloware — CHANGES_REQUESTED** [review]

**Comment by pizzi80** [comment]
@melloware I was not sure if it was better to remove everything or not... it's a 3kLOC single file full of hacks..... now that you told me to remove everything, tomorrow I will continue to work on it 😉😉

**Comment by melloware** [comment]
To me I hate leaving dead blocks of code. That is what Git history is for!

**Pull request closed without merging**